WL-2482

buckett opened the pull request:
Changes that create conversion for existing content in plain text forums, but also improve the filtering we apply to each post.

buckett committed:
Initial work on scanning for URLs

buckett committed:
Fix from Colin to stop matching on emails.

buckett committed:
WL-2482 Improve filtering of HTML -> plain text.

<div>s now get converted to <br>s and we trim multiple <br>s down in 2 so you don't end up with masses of space.
When the contents of a link includes some tags we not detect it to prevent multiple links.
& in links are no longer escaped.

buckett committed:
Markup email addresses as well.

ColinHebert commented on messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/jobs/UpdateMarkupFreeContent.java:
```diff
@@ -0,0 +1,111 @@
+	
+	// These are escaped tags which should have been stripped.
+	// Use non-greedy matcher.
+	private static final Pattern anchorTags = Pattern.compile("<a.*?href=('|\")(.*?)\\1.*?>(.*?)</a>");
```
This regular expression could also capture wrong anchors.

`<a title="hello">blah blah</a><a href="test">test</a>` will be captured only once and will probably result in `<a href="test">test</a>`.

Could that happen?

ColinHebert commented on messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/jobs/UpdateMarkupFreeContent.java:
```diff
@@ -0,0 +1,111 @@
+
+		String clean = messageParsingService.parse(messageParsingService.format(newBody));
+		// Trim multiple <br />
+		clean = clean.replaceAll("<br />(:?\\w*<br />)+", "<br /><br />");
```
Shouldn't it be `\W` and even so wouldn't `\b` or `\s` be more appropriate?

buckett committed:
Fix so we don't scrape up too much when a <a> -href

Eg so <a title="asdsad"> would cause the regexp to carry on scanning for the next href attribute even if it was in a different tag.

buckett committed:
Correctly clean up multiple <br /> tags.

The regexp was just wrong when looking for whitespace.

buckett commented:
Thanks, updated code and added tests.

ColinHebert commented on messageforums-component-impl/src/test/org/sakaiproject/component/app/messageforums/TestMessageParsingServiceImpl.java:
```diff
@@ -62,6 +62,127 @@ public void test4() {
+				messageParsingService.parse("(https://itunes.apple.com/gb/id288751446?mt=8)"));
+	}
+	
+	//@Test
```
Why is this test commented?

buckett replied:
Added reason for test not being run.

buckett committed:
Explain why test isn't being run.

ColinHebert commented:
Merged
